test/controllers/CiapiController/CiapiControllerSpec.scala

```diff
@@ -16,37 +16,62 @@
 
 package controllers.CiapiController
 
+import config.AppConfig
 import controllers.CiapiController.{routes => ciapiRoutes}
+import mocks.MockAuditService
+import models.DAv3AuditModel
 import org.jsoup.Jsoup
 import org.jsoup.nodes.Document
 import org.scalatest.matchers.must.Matchers
 import org.scalatest.wordspec.AnyWordSpecLike
+import play.api.mvc.MessagesControllerComponents
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
+import views.html.CIAPIViews._
+import views.html.CUIViews._
 import views.html.pages.helpers.AppBuilderSpecBase
 
 class CiapiControllerSpec
-  extends AppBuilderSpecBase with Matchers with AnyWordSpecLike {
-
-  private val controller = app.injector.instanceOf[CiapiController]
+  extends AppBuilderSpecBase with Matchers with AnyWordSpecLike with MockAuditService {
+
+  def controller = new CiapiController(
```

Comment:
is there a reason this is a def and not a val?

Reply:
Typically due to safety. Running a def (and generating a new controller for each test) helps prevent cross-test contamination. It's not explicitly required, I can remove it, just a practice I picked up working on teams with complex mocking logic where there was a bigger risk of that happening.

Reply:
I wonder if we specifically want to have one instance of the controller in this case, whilst having multiple tests. Wouldn't this be a closer approximation of the behaviour when running the application?